Explicitly export compile_commands.json and be more paranoid about its existence

krzysz00 opened the pull request.

krzysz00 committed:
Explicitly export compile_commands.json and be more paranoid about its existence

krzysz00 committed:
Put some parentheses on deleteDir, see if that helps

krzysz00 committed:
Revert "Revert "Deduplicate various steps in CI files (#241)" (#250)"

This reverts commit fd2bcca669f5ee7d5df46ec1400d0012e27d6622.

krzysz00 committed:
Merge branch 'generate-compile-commands-in-jenkins-tmp' into generate-compile-commands-in-jenkins

yiqian1 commented:
Can you update Jenkinsfile.perf and Jenkinsfile.perf-gfx908 to use ./llvm/bin/mlir-miopen-driver and ./llvm/bin/mlir-rocm-runner, and run a CI job with Jenkinsfile.perf to make sure it works well?

krzysz00 committed:
Change paths to perf binaries so the test is not broken anymore

krzysz00 committed:
Merge branch 'miopen-dialect' into generate-compile-commands-in-jenkins

krzysz00 committed:
Go to  invocations, this isn't running under bash.

krzysz00 committed:
Merge branch 'miopen-dialect' into generate-compile-commands-in-jenkins

krzysz00 committed:
Minor tweaks to perf Jenkinsfile

krzysz00 committed:
Actually fail on no complie_commands.json not being present.

yiqian1 commented on mlir/utils/jenkins/Jenkinsfile.nightly-perf:
```diff
@@ -48,15 +48,17 @@ pipeline {
+                dir('build') {
+                    // Run MLIR perf benchmarks
+                    sh 'python3 ./llvm/MIOpenDriver.py'
+                }
```
You might want to run this Jenkins job to make sure it works well. I did try dir('build') in my development branch, but for some reason the publishHTML step crashed, though MLIR_vs_MIOpen.html was generated correctly. I don't know why. I'm not very familiar with publishhtml plugins. So I decided to use shell script.

krzysz00 committed:
Per Yi, delete these files, they aren't used anymore

krzysz00 committed:
Steps block in PrettyTable

yiqian1 commented on mlir/utils/jenkins/Jenkinsfile.miopen-test:
```diff
@@ -1,19 +1,79 @@
+                                url: 'https://github.com/ROCmSoftwarePlatform/MIOpen.git'
+                            cmake arguments: "-P install_deps.cmake --minimum",\
+                                installation: "InSearchPath"
+                            cmakeBuild generator: 'Ninja',\
```
MIOpenDriver is not built with Ninja. I found it out because I copied this script to Jenkinsfile.nightly-perf, then MIOpenDriver got seg fault for a few Resnet50 configurations. I changed this piece of code back to 'make', and MIOpenDriver is working well now.

krzysz00 committed:
Merge branch 'miopen-dialect' into generate-compile-commands-in-jenkins

krzysz00 committed:
Switch MIOpen build back to Unix makefiles to prevent crashes

jerryyin commented:
@krzysz00 You should validate all jenkinsfile using the Jenkins::Replay feature. With a record of it passing all CIs, including those not tracked in PR triggered CIs, we can confidently approve/merge your PR. 

The June demo is coming to an end in this week-ish. I want to merge yours very soon once the demo is declared victory.

krzysz00 commented:
Thanks for letting me know about that feature!

krzysz00 commented:
CI tested (the replay build will be linked in once it exists):

- [x] MIOpen dialect ( [replayed build](http://10.236.106.97:8080/blue/organizations/jenkins/MLLIR-MIopen-Resnet50-nightly/detail/MLLIR-MIopen-Resnet50-nightly/34/pipeline/686) )
- [x] Downstream ( [pr build](http://ml-ci.amd.com:21096/job/MLIR/job/mlir/view/change-requests/job/PR-251/4/) )
- [x] Downstream.xdlops ( [pr build](http://10.236.106.97:8080/blue/organizations/jenkins/mlir/detail/PR-251/27/pipeline/31) )
- [x] Nightly perf ( [replayed build](http://ml-ci.amd.com:21096/job/MLIR/job/performance-nightly/10/) )
- [x] Nightly ( [replayed build](http://ml-ci.amd.com:21096/job/MLIR/job/check-mlir-nightly/89/) )
- [x] Nightly.xdlops ( [replayed build](http://10.236.106.97:8080/blue/organizations/jenkins/xdlop-e2e/detail/xdlop-e2e/45/pipeline) )
- [x] Upstream (unused)

krzysz00 committed:
Fix Nightly perf test

krzysz00 committed:
Merge branch 'miopen-dialect' into generate-compile-commands-in-jenkins

krzysz00 committed:
Merge branch 'miopen-dialect' into generate-compile-commands-in-jenkins

whchung commented:
@jerryyin / @yiqian1  now the June demo is completed, please help review this PR.

krzysz00 committed:
Merge branch 'miopen-dialect' into generate-compile-commands-in-jenkins

krzysz00 committed:
Add tracing code for the MIOpen test bug

As discovered with Jerry, we've determined that sometimes the driver
loads system libMIOpen.so, causing intermittent failures. This commit adds
tracing to the Jenkinsfile to catch when this is happening.

krzysz00 committed:
Revert "Add tracing code for the MIOpen test bug" - it's fixed elsewhele.

This reverts commit 7ff0fd31d26e257296a52c52640417f08af7ad3f.

jerryyin commented on mlir/utils/jenkins/Jenkinsfile.downstream:
```diff
@@ -17,66 +20,42 @@ pipeline {
+                cmakeBuild generator: 'Ninja',\
+                    buildDir: 'build',\
+                    buildType: 'Release',\
+                    installation: 'InSearchPath',\
```
[Question] Is this `installations` key/value necessary? What does the `InsearchPath` do on our project?

krzysz00 replied:
The `installations` key is something the CMake plugin uses to determine which CMake to use. InSearchPath means that it uses whatever CMake can be found in $PATH, and so that we control our CMake version via the Dockerfile, as opposed to by defining a particular version of Jenkins to use in the diff.

jerryyin replied:
Okay that makes a lot of sense. I was misunderstanding this as regards to the installation of our llvm project, but turns out this is just cmake build tools related.

jerryyin commented on mlir/utils/jenkins/Jenkinsfile.downstream:
```diff
@@ -17,66 +20,42 @@ pipeline {
+                    steps: [[args: 'check-mlir']],\
+                    cmakeArgs: '''-DMLIR_MIOPEN_DRIVER_ENABLED=1
+                                   -DMLIR_MIOPEN_DRIVER_PR_E2E_TEST_ENABLED=1
+                                   -DCMAKE_EXPORT_COMPILE_COMMANDS=1
```
[Question] I believe the last configuration explicitly ask to export compile commands, but what about the rest of the configurations in the left side of the diff?

krzysz00 replied:
1) It's possible we don't even need this line, since it looks like the root cause of the failure was me using `[[` instead of `[` in a `sh` block
2) The files that aren't `downstream` or `downstream.xdlop` don't run the static checks that need this file.

jerryyin replied:
What about all the other arguments in the original downstream file:

```cmake
-DCMAKE_BUILD_TYPE=Release \
-DLLVM_ENABLE_ASSERTIONS=ON \
-DBUILD_SHARED_LIBS=ON \
-DLLVM_BUILD_LLVM_DYLIB=ON \
-DMLIR_ROCM_RUNNER_ENABLED=1
```

krzysz00 replied:
Those are handled via the fact that we're using the CMakeLists.txt in the root directory and not in ../llvm

jerryyin reviewed (approved):
Thanks for the work!

If your PR repeatedly failing on one config, you may want to either rebase or merge from latest of `miopen-dialect` to pull in a revert.

krzysz00 committed:
Merge branch 'miopen-dialect' into generate-compile-commands-in-jenkins
